### PR TITLE
Components: Use BEM class names to avoid conflict

### LINKED ISSUE #76032: Stats leaks CSS out to other panels
### Quick summary

Some CSS overrides at Stats page (`.promo-card`?) seems to be unscoped, and leaks into Earn page:

After visiting Stats, Earn page's boxes become all wonky and rounded:

Screenshot 2023-04-20 at 16 39 05

https://user-images.githubusercontent.com/87168/233385885-236181b8-440b-409c-bc41-98f9a6450821.mov

### Steps to reproduce

- Open Tools -> Earn
- shift+refresh to ensure you have cache cleared, note square boxes
- Go to Stats
- Come back to earn; note rounded boxes now

### What you expected to happen

Styles not to change.

### What actually happened

Styles leaking.

### Impact

All

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

_No response_

### Logs or notes

@dognose24 @kangzj @jsnmoon hi! I see you've been recently working a lot on Stats with Team Red. Could your team take a look?

### PR DESCRIPTION
**Pull request by jsnmoon**
Fixes #76032.

## Proposed Changes

* Replaces the generic `.promo-card` class name with `.app-promo-card`.
* Prepends all generic class names used by the component with `.app-promo-card__`.

## Testing Instructions

* Ensure that #76032 is no longer reproducible on this PR's live branch.
* Ensure that the story for this component continues to work as expected (`yarn workspace @automattic/components storybook`).
* Ensure that the component continues to work as expected on Stats -> Traffic and Stats -> Insights pages.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/components/src/app-promo-card/index.tsx [hunk 1]
```diff
@@ -87,7 +83,7 @@ export default function AppPromoFullWidth( {
 					components: {
 						a: (
 							<a
-								className="woo"
+								className="app-promo-card__woo-link"
```
Comment (grzegorz-cp) [inline packages/components/src/app-promo-card/index.tsx]: SCSS applies additional styling for `.app-promo-card__link` but the general link class is not added here and below for jetpack.

**Review by grzegorz-cp — CHANGES_REQUESTED** [review]
Great update 👍 . There is only one small thing I noticed.

File: packages/components/src/app-promo-card/index.tsx (hunk 1 above)
Comment (jsnmoon) [reply packages/components/src/app-promo-card/index.tsx]: Ah, good catch. Fixed!

**Review by grzegorz-cp — APPROVED** [review]
👍

**Pull request merged**